src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -341,13 +341,22 @@ object WdlNamespace {
     val callInputSectionErrors = namespace.descendants.collect({ case c: TaskCall => c }).flatMap(
       validateCallInputSection(_, wdlSyntaxErrorFormatter)
     )
+    
+    val workflowOutputErrors = workflows flatMap { _.workflowCalls map { _.calledWorkflow } } collect {
+      case workflow if workflow.workflowOutputWildcards.nonEmpty => 
+        new SyntaxError(
+          s"""Workflow ${workflow.unqualifiedName} is used as a sub workflow but has outputs declared with a deprecated syntax not compatible with sub workflows.
+             |To use this workflow as a sub workflow please update the workflow outputs section to the latest WDL specification.
+             |See https://github.com/broadinstitute/wdl/blob/develop/SPEC.md#outputs""".stripMargin
+        )
+    }
```

Comment:
I wonder if this could work as a forcomp? I could see why it might not and I didn't try it out, so ...